apps/server/src/shared/repo/course/course.repo.integration.spec.ts

```diff
@@ -368,6 +368,54 @@ describe('course repo', () => {
 		});
 	});
 
+	describe('findOneForTeacherOrSubstituteTeacher', () => {
+		const setup = () => {
+			const user = userFactory.buildWithId();
+			const students = userFactory.buildListWithId(2);
```

Comment:
these are not really students, just users. There is no role definition or something like that, is just a basic list of users. To call it students, is a bit misleading.
In fact, when I look at it, the whole setup is just a way to create user(s) for the test. You can have a createUser method, with arg number of users to return.

const createUser = (count: number = 1) => {
  if (count === 1)
     return buildWithId()
  else return buildListWithId